### PR TITLE
Add instructions for using sockets establishing db

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Hi folks,

I didn't find the [GSoD'19 tasks](https://forum.mattermost.org/t/google-season-of-docs-2019/7136) open in any repository, the tasks described on the forum are only internal organisation work. I will be closing few tickets from docs repository or any useful addition I find. Great if you can let me know any open task that I can do in GSoD'19 application period. Thanks.

### CODE REVIEW

**Comment by tapaswenipathak** [comment]
I have signed the CLA.

**Comment by amyblais** [comment]
Hi @tapaswenipathak,

Thank you for this contribution!

A great way to get involved is by choosing tickets from the docs repository or by adding any new doc pages (just like you mentioned). You can also join our Season of Docs channel on our [Community server](https://community.mattermost.com/core/channels/season-of-docs) where you can let our product managers know if you have any questions.

For this PR, would you be able to add this new doc section to the table of contents: https://docs.mattermost.com/guides/administrator.html#installing-mattermost?

File: source/guides/administrator.rst [hunk 1]
```diff
@@ -31,6 +31,7 @@ Learn how to get Mattermost running on your environment.
    Installing on RHEL 6.6 </install/install-rhel-66.rst>
    Installing on RHEL 7.1 </install/install-rhel-71.rst>
    Installing on CentOS, Oracle Linux, and Scientific Linux </install/install-centos-oracle-scientific.rst>
+   Instructions on setting up a socket based mattermost db </install/sockets-db.rst>
```
Comment (amyblais) [inline source/guides/administrator.rst]:
```suggestion
   Instructions on setting up a socket-based Mattermost database </install/sockets-db.rst>
```

File: source/install/sockets-db.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
+Using Sockets For DB
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
Using Sockets for Database
```

File: source/install/sockets-db.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+====================
+
+Mattermost requires a database back-end. If you plan running it on the machine,
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
Mattermost requires a database back-end. If you plan to run it on the machine,
```

File: source/install/sockets-db.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+install MySQL or PostgreSQL as database. In this document let's understand how
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
install MySQL or PostgreSQL as the database. In this document let's understand how
```

File: source/install/sockets-db.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+you can use sockets for setting up db.
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
you can use sockets for setting up the database.
```

File: source/install/sockets-db.rst [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    $ mysql -u root -p
+    CREATE DATABASE mattermostdb;
+    CREATE USER mmuser IDENTIFIED BY 'mmuser_password';
+    GRANT ALL ON mattermostdb.* TO mmuser;
+
+
+PostgreSQL
+----------
+
+- Install and configure postgres SQL.
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Install and configure PostgreSQL.
```

File: source/install/sockets-db.rst [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+With TCP socket
+---------------
+
+- Create the new user while connecting to the server as `postgres` user
+  (you will be prompted for a password for the new user):
+
+   ``sudo -u postgres createuser -P mmuser``
+
+- Create the Mattermost database, owned by `mmuser` user:
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Create the Mattermost database, owned by ``mmuser`` user:
```

File: source/install/sockets-db.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+PostgreSQL
+----------
+
+- Install and configure postgres SQL.
+- Choose between TCP or UNIX Socket, and jump to the corresponding section.
+
+With TCP socket
+---------------
+
+- Create the new user while connecting to the server as `postgres` user
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Create the new user while connecting to the server as ``postgres`` user
```

File: source/install/sockets-db.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  (you will be prompted for a password for the new user):
+
+   ``sudo -u postgres createuser -P mmuser``
+
+- Create the Mattermost database, owned by `mmuser` user:
+
+  ``sudo -u postgres createdb -O mmuser mattermostdb``
+
+- In the connections and authentications section, set the listen_address list
+  line to your needs:
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
  line per your needs:
```

File: source/install/sockets-db.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+With Unix socket
+----------------
+
+- Create the new user while connecting to the server as ``postgres`` user:
+
+  .. code-block:: bash
+
+     sudo -u postgres createuser mattermost
+
+- Create the Mattermost database, owned by `mattermost` user:
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Create the Mattermost database, owned by ``mattermost`` user:
```

File: source/install/sockets-db.rst [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+  .. code-block:: bash
+
+     sudo -u postgres createuser mattermost
+
+- Setup the Unix socket by adding the following line to ``/var/lib/postgres/data/pg_hba.conf:``
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Setup the Unix socket by adding the following line to ``/var/lib/postgres/data/pg_hba.conf``:
```

File: source/install/sockets-db.rst [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+- Mattermost is configured in ``/etc/webapps/mattermost/config.json``.
+  Strings need to be quoted.
+
+- The ``DriverName`` setting: ``mysql`` for MySQL and ``postgres`` for PostgreSQL.
+
+  The ``DataSource``:
+
+  - For MySQL, set it to ``mmuser:mmuser_password@unix(/run/mysqld/mysqld.sock)/mattermostdb?charset=utf8mb4,utf8``.
+  - For PostgreSQL
+    - TCP socket: ``postgres://mmuser:mmuser_password@127.0.0.1:5432/mattermostdb?sslmode=disable&connect_timeout=10``    - Unix socket: ``postgres:///mattermostdb?host=/run/postgresql``, mattermostdb is the name of the database and /run/postgresql is the directory containing the Unix socket
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
    - TCP socket: ``postgres://mmuser:mmuser_password@127.0.0.1:5432/mattermostdb?sslmode=disable&connect_timeout=10``    
    - Unix socket: ``postgres:///mattermostdb?host=/run/postgresql``, where ``mattermostdb`` is the name of the database and ``/run/postgresql`` is the directory containing the Unix socket.
```

File: source/install/sockets-db.rst [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+- Create the new user while connecting to the server as `postgres` user
+  (you will be prompted for a password for the new user):
+
+   ``sudo -u postgres createuser -P mmuser``
+
+- Create the Mattermost database, owned by `mmuser` user:
+
+  ``sudo -u postgres createdb -O mmuser mattermostdb``
+
+- In the connections and authentications section, set the listen_address list
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- In the connections and authentications section, set the ``listen_address`` list
```

File: source/install/sockets-db.rst [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+- Then add a line like the following to the authentication config:
+
+  .. code-block:: bash
+
+     /var/lib/postgres/data/pg_hba.conf
+     # TYPE  DATABASE        USER            ADDRESS                 METHOD
+     # IPv4 local connections:
+     host    all             all             ip_address/32   md5
+
+- Run the setup using
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Run the setup using:
```

File: source/install/sockets-db.rst [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+- Setup the Unix socket by adding the following line to ``/var/lib/postgres/data/pg_hba.conf:``
+
+  .. code-block:: bash
+
+     local    mattermostdb    mattermost    peer
+
+- Restart postgresql.service.
+
+- Run the setup using
```
Comment (amyblais) [inline source/install/sockets-db.rst]:
```suggestion
- Run the setup using:
```

**Comment by amyblais** [comment]
Hi @tapaswenipathak, I just added suggestions for small tweaks for readability. If you agree and apply the changes, I'll then give this to a developer to review as well. Let me know any questions,

**Comment by tapaswenipathak** [comment]
@amyblais sure! I committed the edits, thanks.

**Pull request merged**